app/controllers/api/v3/courses_controller.rb

```diff
@@ -24,11 +28,21 @@ def show
 
     private
 
+      def allowed_to_disable_pagination?
+        fields_for_sitemap?
+      end
+
+      def fields_for_sitemap?
+        if params[:fields] && (courses = params[:fields][:courses])
+          (courses.split(",") & %w[course_code provider_code changed_at]).size == 3
+        end
+      end
+
       def build_courses
         @courses = if @provider.present?
-                     @provider.courses
+                     @provider.courses.includes(:provider)
```

Comment:
Just wondering why?

All the courses here will have the same `@provider` aka `@provider.courses.first.provider == @provider`

Reply:
The same request made http://localhost:3001/api/v3/recruitment_cycles/2020/providers/L26/courses?fields[courses]=course_code,provider_code,changed_at

With my above change with eager load
```
2020-03-05 09:15:44.837890 D [77321:puma threadpool 001] Rack -- Started -- { :method => "GET", :path => "/api/v3/recruitment_cycles/2020/providers/L26/courses?fields[courses]=course_code,provider_code,changed_at", :ip => "::1" }
2020-03-05 09:15:44.849744 D [77321:puma threadpool 001] API::V3::CoursesController -- Processing #index
2020-03-05 09:15:44.852234 D [77321:puma threadpool 001] (0.765ms) ActiveRecord -- RecruitmentCycle Load -- { :sql => "SELECT \"recruitment_cycle\".* FROM \"recruitment_cycle\" WHERE \"recruitment_cycle\".\"year\" = $1 LIMIT $2", :binds => { :year => "2020", :limit => 1 } }
2020-03-05 09:15:44.856626 D [77321:puma threadpool 001] (1.593ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"recruitment_cycle_id\" = $1 AND \"provider\".\"discarded_at\" IS NULL AND \"provider\".\"provider_code\" = $2 LIMIT $3", :binds => { :recruitment_cycle_id => 2, :provider_code => "L26", :limit => 1 } }
2020-03-05 09:15:44.865165 D [77321:puma threadpool 001] (3.510ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL AND \"course\".\"id\" IN (SELECT \"course_site\".\"course_id\" FROM \"course_site\" WHERE \"course_site\".\"status\" = $2 AND \"course_site\".\"publish\" = $3)", :binds => { :provider_id => 13732, :status => "R", :publish => "Y" } }
2020-03-05 09:15:44.866185 D [77321:puma threadpool 001] (0.032ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL AND \"course\".\"id\" IN (SELECT \"course_site\".\"course_id\" FROM \"course_site\" WHERE \"course_site\".\"status\" = $2 AND \"course_site\".\"publish\" = $3)", :binds => { :provider_id => 13732, :status => "R", :publish => "Y" } }
2020-03-05 09:15:44.871635 D [77321:puma threadpool 001] (3.503ms) ActiveRecord -- Course Load -- { :sql => "SELECT \"course\".* FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL AND \"course\".\"id\" IN (SELECT \"course_site\".\"course_id\" FROM \"course_site\" WHERE \"course_site\".\"status\" = $2 AND \"course_site\".\"publish\" = $3) LIMIT $4 OFFSET $5", :binds => { :provider_id => 13732, :status => "R", :publish => "Y", :limit => 100, :offset => 0 } }
2020-03-05 09:15:44.875284 D [77321:puma threadpool 001] (1.213ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1", :binds => { :id => 13732 } }
2020-03-05 09:15:44.878614 I [77321:puma threadpool 001] Rails -- Completed JSON API rendering (11.84ms)
2020-03-05 09:15:44.878875 I [77321:puma threadpool 001] (28.9ms) API::V3::CoursesController -- Completed #index -- { :controller => "API::V3::CoursesController", :action => "index", :params => { "fields" => { "courses" => "course_code,provider_code,changed_at" }, "recruitment_cycle_year" => "2020", "provider_code" => "L26" }, :format => "HTML", :method => "GET", :path => "/api/v3/recruitment_cycles/2020/providers/L26/courses", :status => 200, :view_runtime => 7.73, :db_runtime => 10.62, :status_message => "OK" }
```

The original before my change without eager load
```
2020-03-05 09:16:06.053493 D [77321:puma threadpool 002] Rack -- Started -- { :method => "GET", :path => "/api/v3/recruitment_cycles/2020/providers/L26/courses?fields[courses]=course_code,provider_code,changed_at", :ip => "::1" }
2020-03-05 09:16:06.206082 D [77321:puma threadpool 002] API::V3::CoursesController -- Processing #index
2020-03-05 09:16:06.221184 D [77321:puma threadpool 002] (0.325ms) ActiveRecord -- RecruitmentCycle Load -- { :sql => "SELECT \"recruitment_cycle\".* FROM \"recruitment_cycle\" WHERE \"recruitment_cycle\".\"year\" = $1 LIMIT $2", :binds => { :year => "2020", :limit => 1 } }
2020-03-05 09:16:06.244622 W [77321:puma threadpool 002] Provider -- An enum element in Provider uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.245346 W [77321:puma threadpool 002] Provider -- An enum element in Provider uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.280903 D [77321:puma threadpool 002] (0.595ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"recruitment_cycle_id\" = $1 AND \"provider\".\"discarded_at\" IS NULL AND \"provider\".\"provider_code\" = $2 LIMIT $3", :binds => { :recruitment_cycle_id => 2, :provider_code => "L26", :limit => 1 } }
2020-03-05 09:16:06.337544 W [77321:puma threadpool 002] Course -- An enum element in Course uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.337987 W [77321:puma threadpool 002] Course -- An enum element in Course uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.339423 W [77321:puma threadpool 002] Course -- An enum element in Course uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.339727 W [77321:puma threadpool 002] Course -- An enum element in Course uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.341004 W [77321:puma threadpool 002] Course -- An enum element in Course uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.341182 W [77321:puma threadpool 002] Course -- An enum element in Course uses the prefix 'not_'. This will cause a conflict with auto generated negative scopes.
2020-03-05 09:16:06.393674 D [77321:puma threadpool 002] (0.693ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL AND \"course\".\"id\" IN (SELECT \"course_site\".\"course_id\" FROM \"course_site\" WHERE \"course_site\".\"status\" = $2 AND \"course_site\".\"publish\" = $3)", :binds => { :provider_id => 13732, :status => "R", :publish => "Y" } }
2020-03-05 09:16:06.394487 D [77321:puma threadpool 002] (0.028ms) ActiveRecord -- { :sql => "SELECT COUNT(*) FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL AND \"course\".\"id\" IN (SELECT \"course_site\".\"course_id\" FROM \"course_site\" WHERE \"course_site\".\"status\" = $2 AND \"course_site\".\"publish\" = $3)", :binds => { :provider_id => 13732, :status => "R", :publish => "Y" } }
2020-03-05 09:16:06.421026 D [77321:puma threadpool 002] (1.922ms) ActiveRecord -- Course Load -- { :sql => "SELECT \"course\".* FROM \"course\" WHERE \"course\".\"provider_id\" = $1 AND \"course\".\"discarded_at\" IS NULL AND \"course\".\"id\" IN (SELECT \"course_site\".\"course_id\" FROM \"course_site\" WHERE \"course_site\".\"status\" = $2 AND \"course_site\".\"publish\" = $3) LIMIT $4 OFFSET $5", :binds => { :provider_id => 13732, :status => "R", :publish => "Y", :limit => 100, :offset => 0 } }
2020-03-05 09:16:06.447264 D [77321:puma threadpool 002] (0.312ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.447961 D [77321:puma threadpool 002] (0.013ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.448365 D [77321:puma threadpool 002] (0.012ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.448748 D [77321:puma threadpool 002] (0.011ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.449123 D [77321:puma threadpool 002] (0.012ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.449497 D [77321:puma threadpool 002] (0.011ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.449875 D [77321:puma threadpool 002] (0.011ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.450367 D [77321:puma threadpool 002] (0.019ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.451294 D [77321:puma threadpool 002] (0.022ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.451955 D [77321:puma threadpool 002] (0.027ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.452839 D [77321:puma threadpool 002] (0.031ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.453715 D [77321:puma threadpool 002] (0.023ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.454411 D [77321:puma threadpool 002] (0.020ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.455207 D [77321:puma threadpool 002] (0.025ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.455776 D [77321:puma threadpool 002] (0.016ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.456394 D [77321:puma threadpool 002] (0.018ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.456909 D [77321:puma threadpool 002] (0.015ms) ActiveRecord -- Provider Load -- { :sql => "SELECT \"provider\".* FROM \"provider\" WHERE \"provider\".\"id\" = $1 LIMIT $2", :binds => { :id => 13732, :limit => 1 } }
2020-03-05 09:16:06.458603 I [77321:puma threadpool 002] Rails -- Completed JSON API rendering (41.82ms)
2020-03-05 09:16:06.458897 I [77321:puma threadpool 002] (252.7ms) API::V3::CoursesController -- Completed #index -- { :controller => "API::V3::CoursesController", :action => "index", :params => { "fields" => { "courses" => "course_code,provider_code,changed_at" }, "recruitment_cycle_year" => "2020", "provider_code" => "L26" }, :format => "HTML", :method => "GET", :path => "/api/v3/recruitment_cycles/2020/providers/L26/courses", :status => 200, :view_runtime => 37.8, :db_runtime => 23.83, :status_message => "OK" }
```

Reply:
cool